src/jdk.compiler/share/classes/com/sun/tools/javac/code/Types.java

```diff
@@ -3613,10 +3611,10 @@ public Type visitTypeVar(TypeVar t, Void ignored) {
                     return to.head.withTypeVar(t);
                 }
                 if (allowUniversalTVars &&
-                        t.createdFromUniversalTypeVar &&
+                        !t.isValueProjection() &&
```

Comment:
Not sure about this here - "not value projection might also mean it's a regular tvar". Doesn't this test want to check for `T.ref` (in which case you should check for `isReferenceProjection`